versions/development/SPEC.md

```diff
@@ -1218,81 +1270,316 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
   }
 }
 ```
 
-Most key/value pairs are arbitrary.  However, the following keys have recommended conventions:
 
-#### docker
+#### cpu (**Required**)
 
-Location of a Docker image for which this task ought to be run.  This can have a format like `ubuntu:latest` or `broadinstitute/scala-baseimage` in which case it should be interpreted as an image on DockerHub (i.e. it is valid to use in a `docker pull` command).
+The `cpu` key defines the _minimum_ CPU required for this task, which must be available prior to the engine starting execution. The engine does not need to provide the exact amount of CPU requested (depending on the underlying infrastructure restrictions), however it may ONLY provision more CPU than requested and not less. For example if the wdl requested `cpu: 0.5`, but only discrete values were supported, then the engine might choose to provision `1.0` cpu instead.  Values are expected to be a `Float` or an `Int`
 
 ```wdl
-task docker_test {
-  input {
-    String arg
-  }
-  command {
-    python process.py ${arg}
-  }
+task cpu_example {
+  #....	
   runtime {
-    docker: "ubuntu:latest"
+    cpu: 8
   }
 }
+
 ```
 
-#### memory
+#### memory (**Required**)
 
-Memory requirements for this task.  Two kinds of values are supported for this attributes:
+The `memory` key defines the _minimum_ memory required for this task which must be available prior to the engine starting execution. The engine does not need to provide the exact amount of memory requested, however it may ONLY provision more memory then requested and not less. For example, if the wdl requested `1 GB` but only blocks of `4 GB` were available, the engine might choose to provision `4.0 GB` instead.  Two kinds of values are supported for this attribute:
 
 * `Int` - Interpreted as bytes
 * `String` - This should be a decimal value with suffixes like `B`, `KB`, `MB` or binary suffixes `KiB`, `MiB`.  For example: `6.2 GB`, `5MB`, `2GiB`.
 
 ```wdl
 task memory_test {
-  input {
-    String arg
+  #....
+  runtime {
+    memory: "2 GB"
   }
+}
+```
 
-  command {
-    python process.py ${arg}
+#### gpu (**Optional**)
+
+The `gpu` key provides a way to accommodate modern workflows which are increasingly becoming reliant on GPU computations. This key is a `true` or `false` value and simply indicates to the engine that a task requires a GPU to run to completion. A task with this flag set to `true` is guaranteed to only run if a GPU is a available within the runtime environment. It is the responsibility of the engine to check prior to execution whether a GPU is provisionable, and if not, preemptively fail the task.
+
+It is important to note, that this flag does not provide any information on the quantity or type of GPUs to make available to the task. This information should be provided within the `hints` section according to the specific engine parameters.
+
+
+```wdl
+task gpu_test {
+  #.....
+  runtime {
+    gpu: true
   }
+}
+```
+
+
```

Comment:
I suggest having a default flag of `false` for this field. Otherwise, we would be a making tasks more verbose than necessary.